### DISCUSSION TITLE
Javascript working on all pages?

### DISCUSSION THREAD

**Discussion by JustSxm** [vercel/next.js#22164] [Help] [answered]
Hello!

I'm really new to Next.js and as I've been using it to make a website where I'm trying stuff I've discovered that my javascript will run even on page navigation (I suppose because it is pre-rendered?) 

Is there a way where I can prevent that since I've made the javascript for one specific page only, not all of them. Thank you!

**Comment by michrome** [comment]
Hi @JustSxm !

When pre-rendering, Next.js generates HTML for each page in advance, instead of having it all done by client-side JavaScript.

Pre-rendered pages are still reactive: when a page is loaded by the browser, its JavaScript code runs and makes the page fully interactive (this process is called hydration). Generated HTML is associated with minimal JavaScript code necessary for that page.

---

That said, if you _really_ want to disable the runtime JavaScript and _all the goodness_ it brings, there is an **unstable** config you can try:

```javascript
// pages/my-page.js

export const config = {
  unstable_runtimeJS: false,
};

function MyPage() {
  return <h1>Hello World</h1>;
}

export default MyPage;

```

**Reply by JustSxm** [reply]
> Hi @JustSxm !
> 
> When pre-rendering, Next.js generates HTML for each page in advance, instead of having it all done by client-side JavaScript.
> 
> Pre-rendered pages are still reactive: when a page is loaded by the browser, its JavaScript code runs and makes the page fully interactive (this process is called hydration). Generated HTML is associated with minimal JavaScript code necessary for that page.
> 
> That said, if you _really_ want to disable the runtime JavaScript and _all the goodness_ it brings, there is an **unstable** config you can try:
> 
> ```js
> // pages/my-page.js
> 
> export const config = {
>   unstable_runtimeJS: false,
> };
> 
> function MyPage() {
>   return <h1>Hello World</h1>;
> }
> 
> export default MyPage;
> ```

Hello! Well, I would prefer to not disable it at a whole, here is my problem:
  <Head>
        <title>Sam</title>
        <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.5.1/jquery.min.js"></script>
        <script type="text/javascript" src="./scroll.js"></script>
        <script type="text/javascript" src="./scrollAnimation.js"></script>
      </Head>
      
      I'm using two javascript external scripts that I've made. Though, they are made for one page. So i'd like those script to be stopped from executing or simply not be executed on other pages

**Comment by JustSxm** [comment, marked as answer]
I've answered my own question(guess I learned at the same time the use of </Link> !)
I was using next.js <Link> for the link and apparently this keeps the javascript loaded when navigating, that's my bad!
